core/drivers/atmel_wdt.c

```diff
@@ -11,12 +11,10 @@
 #include <kernel/delay.h>
 #include <kernel/dt.h>
 #include <kernel/dt_driver.h>
-#include <kernel/interrupt.h>
 #include <kernel/pm.h>
 #include <matrix.h>
 #include <mm/core_mmu.h>
 #include <platform_config.h>
-#include <tee_api_types.h>
```

Comment:
Removing these #include should be a separate commit

Reply:
`#include <kernel/interrupt.h>` should be removed in this commit also

Reply:
Do you mean the `kernel/interrupt.h` in [core/include/drivers/wdt.h](https://github.com/OP-TEE/optee_os/blob/63bfec5e2/core/include/drivers/wdt.h#L10)? As interrupt is used in [this file](https://github.com/OP-TEE/optee_os/blob/63bfec5e264b66c00bfc1b6ba807ec0f75495891/core/drivers/atmel_wdt.c#L252) the header can not be removed.
Updated with removing some other header files, thanks.

Reply:
`#include <tee_api_types.h>` is needed for `TEE_Result` definition.
`#include <kernel/delay.h>` is needed for `udelay()` declaration.
`#include <kernel/dt_driver.h>` is needed for `DEFINE_DT_DRIVER()` definition.

Reply:
I forget to save many other comments when posting my review outcome. Here they are.

Many of these header files are needed IMO:
`#include <kernel/delay.h>` is needed for `udelay()` declaration.
`#include <kernel/dt.h>` is needed for `dt_map_dev()` and `fdt_get_status()` declaration and some other macros.
`#include <kernel/dt_driver.h>` is needed for `DEFINE_DT_DRIVER()` and few other definitions/declarations.
`#include <mm/core_mmu.h>` is needed for `core_mmu_remove_mapping()` declaration.
`#include <tee_api_types.h>` could be omitted but _tee_api_defines.h_ is needed for definition of `TEE_SUCCESS` and `TEE_ERROR_*`.

`#include <kernel/interrupt.h>` is needed for many function declarations and types definitions.

There are some header files that are missing: _util.h_, _stdbool.h_, _stdint.h_, _kernel/panic.h_, _trace.h_, _malloc.h_.

Reply:
Update `#include list of the header files by:
 - Remove the unused header files from "#include".
 - `#include` the header files needed explicitly even if they are included indirectly.